engine-standalone-storage/src/relayer_db/types.rs

```diff
@@ -146,12 +193,47 @@ pub struct TransactionRow {
     pub output: Vec<u8>,
 }
 
-impl From<postgres::Row> for TransactionRow {
-    fn from(row: postgres::Row) -> Self {
-        let block: i64 = row.get("block");
+struct TransactionRowBlock(pub i64);
+
+impl TryFrom<TransactionRowBlock> for u64 {
+    type Error = TryFromIntError;
+
+    fn try_from(value: TransactionRowBlock) -> Result<Self, Self::Error> {
+        // set negative values to 0
+        value.0.max(0).try_into()
+    }
+}
+
+struct TransactionRowIndex(pub i32);
+
+impl TryFrom<TransactionRowIndex> for u16 {
+    type Error = TryFromIntError;
+
+    fn try_from(value: TransactionRowIndex) -> Result<Self, Self::Error> {
+        // set Maximum value to u16::MAX
+        value.0.min(u16::MAX.into()).try_into()
+    }
+}
+
+struct TransactionRowId(pub i64);
+
+impl TryFrom<TransactionRowId> for u64 {
+    type Error = TryFromIntError;
+
+    fn try_from(value: TransactionRowId) -> Result<Self, Self::Error> {
+        // set negative values to 0
+        value.0.max(0).try_into()
+    }
+}
+
+impl TryFrom<postgres::Row> for TransactionRow {
+    type Error = postgres::Error;
+
+    fn try_from(row: postgres::Row) -> Result<Self, Self::Error> {
+        let block: TransactionRowBlock = TransactionRowBlock(row.get("block"));
         let block_hash = get_hash(&row, "block_hash");
-        let index: i32 = row.get("index");
-        let id: i64 = row.get("id");
+        let index: TransactionRowIndex = TransactionRowIndex(row.get("index"));
```

Comment:
It's needless to define explicit type here because you use it on the right side.
```suggestion
        let index = TransactionRowIndex(row.get("index"));
```